### PR TITLE
Update FedProx baseline

### PR DESCRIPTION
**Pull request by jafermarq**
After the `FedProx` baseline was added in the SoR format, some formatting checks and tests were introduced. Those were never ran against `FedProx`. This PR ensures tests pass. To achieve this:
 * The `pyproject.toml` has been updated with the new content (that already is part of that in the [EXTENDED_README.md](https://github.com/adap/flower/tree/main/baselines/baseline_template) in the template baseline)
 * Then, the formatting script and tests were run and pass (except one --> see comment below)
 * Fixed a few typos

Once this is merged, progress can continue with https://github.com/adap/flower/pull/2210 and get it merged.

### CODE REVIEW

**Comment by jafermarq** [comment]
[This line](https://github.com/adap/flower/blob/b88569a66ffc1965d64227c45cb51bfc292d8934/baselines/fedprox/fedprox/main.py#L77) will trigger a mypy error since back in flwr=1.3 (the version pinned for SoR) `start_simulation` expected the `client_fn` callable to return a `Client` (not including `NumpyClient`). @danieljanes, what do you suggest doing here ? Should we bump flwr to 1.5 in `pyproject.toml` in a separate PR?

**Comment by jafermarq** [comment]
Will update once https://github.com/adap/flower/pull/2301 is merged

**Comment by jafermarq** [comment]
needs change from #2307

File: baselines/fedprox/fedprox/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+"""Fedprox package."""
```
Comment (danieljanes) [inline baselines/fedprox/fedprox/__init__.py:1]:
```suggestion
"""FedProx package."""
```

File: baselines/fedprox/fedprox/main.py [hunk 2]
```diff
@@ -10,17 +12,18 @@
 from fedprox.dataset import load_datasets
 from fedprox.utils import save_results_as_pickle
 
+FitConfig = Dict[str, Union[bool, float]]
+
 
 @hydra.main(config_path="conf", config_name="config", version_base=None)
 def main(cfg: DictConfig) -> None:
-    """Main function to run CNN federated learning on MNIST.
+    """Rrun CNN federated learning on MNIST.
```
Comment (danieljanes) [inline baselines/fedprox/fedprox/main.py:20]:
```suggestion
    """Run CNN federated learning on MNIST.
```

File: baselines/fedprox/fedprox/server.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+"""Flower Server."""
 from collections import OrderedDict
```
Comment (danieljanes) [inline baselines/fedprox/fedprox/server.py:2]:
```suggestion
"""Flower Server."""


from collections import OrderedDict
```

File: baselines/fedprox/fedprox/strategy.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+"""Flower strategy."""
 from typing import List, Tuple, Union
```
Comment (danieljanes) [inline baselines/fedprox/fedprox/strategy.py:2]:
```suggestion
"""Flower strategy."""


from typing import List, Tuple, Union
```

**Comment by jafermarq** [comment]
needs https://github.com/adap/flower/pull/2327

File: baselines/fedprox/fedprox/strategy.py [hunk 5]
```diff
@@ -37,16 +39,17 @@ def aggregate_fit(
...
-        stragglers in this round."""
+        """Discard all the models sent by the clients that were stragglers.
 
+        in this round.
+        """
         # Record which client was a straggler in this round
         stragglers_mask = [res.metrics["is_straggler"] for _, res in results]
 
-        print(f"Num stragglers in round: {sum(stragglers_mask)}")
+        # print(f"Num stragglers in round: {sum(stragglers_mask)}")
```
Comment (charlesbvll) [inline baselines/fedprox/fedprox/strategy.py:49]: Should we remove this line altogether?

**Review by charlesbvll — APPROVED** [review]
LGTM!

File: baselines/fedprox/fedprox/main.py [hunk 6]
```diff
@@ -53,7 +56,9 @@ def main(cfg: DictConfig) -> None:
     def get_on_fit_config():
         def fit_config_fn(server_round: int):
             # resolve and convert to python dict
-            fit_config = OmegaConf.to_container(cfg.fit_config, resolve=True)
+            fit_config: FitConfig = OmegaConf.to_container(  # type: ignore
+                cfg.fit_config, resolve=True
+            )
             fit_config["curr_round"] = server_round  # add round info
```
Comment (danieljanes) [inline baselines/fedprox/fedprox/main.py:62]: Do we want to convert the `server_round` `int` to `float`, or rather change the type definition to include `int` as a possible value type?

File: baselines/fedprox/pyproject.toml [hunk 7]
```diff
@@ -37,21 +37,23 @@ classifiers = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.8.15, <3.12.0"
-flwr = "1.3.0"
-ray = "1.11.1"
+python = ">=3.10.0, <3.11.0"
```
Comment (danieljanes) [inline baselines/fedprox/pyproject.toml:40]: Do we want to exclude 3.11?

File: doc/source/ref-changelog.md [hunk 8]
```diff
@@ -8,6 +8,10 @@
 
 - **General updates to baselines** ([#2301](https://github.com/adap/flower/pull/2301).[#2305](https://github.com/adap/flower/pull/2305), [#2307](https://github.com/adap/flower/pull/2307), [#2327](https://github.com/adap/flower/pull/2327))
 
+- **New or updates baselines**
```
Comment (danieljanes) [inline doc/source/ref-changelog.md:11]:
```suggestion
- **Update Flower Baselines**
```

File: baselines/fedprox/fedprox/main.py (hunk 6 above)
Comment (jafermarq) [reply baselines/fedprox/fedprox/main.py:62]: maybe in another pr another time

File: baselines/fedprox/pyproject.toml (hunk 7 above)
Comment (jafermarq) [reply baselines/fedprox/pyproject.toml:40]: is not useful if the best we can do to specify some (more complicated) packages such as `torch` need to pin the point to a specific wheel+python version. Maybe there is a better way of doing this, but i couldn't find one:
```
torch = { url = "https://download.pytorch.org/whl/cu117/torch-2.0.1%2Bcu117-cp310-cp310-linux_x86_64.whl"}
torchvision = { url = "https://download.pytorch.org/whl/cu117/torchvision-0.15.2%2Bcu117-cp310-cp310-linux_x86_64.whl"}
```

Comment (charlesbvll) [reply baselines/fedprox/pyproject.toml:40]: Should we move this to another PR?

Comment (jafermarq) [reply baselines/fedprox/pyproject.toml:40]: yes. let's leave it as is currently in the `.toml` then?

**Review by danieljanes — APPROVED** [review]

**Pull request merged**